src/Microsoft.PowerShell.Security/security/CertificateProvider.cs

```diff
@@ -2991,28 +3092,33 @@ public string LocationName
         {
             get
             {
-                return _location.ToString();
+                return CertLocation.ToString();
             }
         }
 
         /// <summary>
         /// Gets the location as a
         /// <see cref="System.Security.Cryptography.X509Certificates.StoreLocation"/>
+        /// The property CertLocation should be used instead of this one as it
+        /// supports store locations other than CurrentUser and LocalMachine.
         /// </summary>
-        public StoreLocation Location
+        public StoreLocation? Location
         {
             get
             {
-                return _location;
+                return (int)CertLocation < 3 ? (StoreLocation)CertLocation : null;
```

Comment:
Would it be better to use `Enum.IsDefined()`?

Reply:
Yep, wasn't aware of this API, will update it.